httpd/http_server.c

```diff
@@ -0,0 +1,333 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "daemon/common.h"
+#include "http_server.h"
+#include "h2o.h"
+
+#include "h2o_utils.h"
+
+static h2o_globalconf_t config;
+static h2o_context_t ctx;
+static h2o_accept_ctx_t accept_ctx;
+
+#define CONTENT_JSON_UTF8 H2O_STRLIT("application/json; charset=utf-8")
+#define CONTENT_TEXT_UTF8 H2O_STRLIT("text/plain; charset=utf-8")
+#define NBUF_INITIAL_SIZE_RESP (4096)
+#define API_V1_PREFIX "/api/v1/"
+#define HOST_SELECT_PREFIX "/host/"
+
+#define HTTPD_CONFIG_SECTION "httpd"
+#define HTTPD_ENABLED_DEFAULT false
+
+static void on_accept(uv_stream_t *listener, int status)
+{
+    uv_tcp_t *conn;
+    h2o_socket_t *sock;
+
+    if (status != 0)
+        return;
+
+    conn = h2o_mem_alloc(sizeof(*conn));
+    uv_tcp_init(listener->loop, conn);
+
+    if (uv_accept(listener, (uv_stream_t *)conn) != 0) {
+        uv_close((uv_handle_t *)conn, (uv_close_cb)free);
+        return;
+    }
+
+    sock = h2o_uv_socket_create((uv_stream_t *)conn, (uv_close_cb)free);
+    h2o_accept(&accept_ctx, sock);
+}
+
+static int create_listener(const char *ip, int port)
+{
+    static uv_tcp_t listener;
+    struct sockaddr_in addr;
+    int r;
+
+    uv_tcp_init(ctx.loop, &listener);
+    uv_ip4_addr(ip, port, &addr);
+    if ((r = uv_tcp_bind(&listener, (struct sockaddr *)&addr, 0)) != 0) {
+        fprintf(stderr, "uv_tcp_bind:%s\n", uv_strerror(r));
+        goto Error;
+    }
+    if ((r = uv_listen((uv_stream_t *)&listener, 128, on_accept)) != 0) {
+        fprintf(stderr, "uv_listen:%s\n", uv_strerror(r));
+        goto Error;
+    }
+
+    return 0;
+Error:
+    uv_close((uv_handle_t *)&listener, NULL);
+    return r;
+}
+
+static int ssl_init()
+{
+    if (!config_get_boolean(HTTPD_CONFIG_SECTION, "ssl", false))
```

Comment:
When you cannot get the `SSL` certificate from `HTTPD` section, I suggest to also test `WEB` section. This will keep compatibility with old versions.

Reply:
That is indeed the future plan. When we are about to remove internal web-server I want new one to read as fallback the old web-server configuration (to avoid disruption). The new `[httpd]` section motive is to clean up the config but yes we will read `[web]` section for backwards compatibility in future. Currently it is not done this way because the servers are independent and can run side to side.

Reply:
until that moment i would prefer the configurations to be separate and explicit to avoid surprise of users

Reply:
All right, thanks to explain!